src/main/java/net/minestom/server/event/Event.java

```diff
@@ -1,11 +1,94 @@
 package net.minestom.server.event;
 
 import net.minestom.server.event.handler.EventHandler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.function.Consumer;
+import java.util.function.Function;
 
 /**
  * Object which can be listened to by an {@link EventHandler}.
  * <p>
  * Called using {@link EventHandler#callEvent(Class, Event)}.
  */
 public class Event {
+
+    public static <T extends EntityEvent> Listener.Builder<T> entity(Class<T> eventType) {
+        return new Listener.Builder<>(eventType);
+    }
+
+    public static <T extends PlayerEvent> Listener.Builder<T> player(Class<T> eventType) {
+        return new Listener.Builder<>(eventType);
+    }
+
+
+    public static class Listener<T extends Event> {
```

Comment:
This should probably end up in its own class at some point, especially if we end up adding an expiry/cancellation system.

Reply:
Perhaps, a bit afraid that it will end up confusing with it and `EventCallback`